### ISSUE TITLE
Query about train-test split

### ISSUE THREAD

**Issue by supriya-gdptl** [BigRedT/vico#1]
Thanks @BigRedT, for sharing the code. 
I have a question regarding multi-task log-bilinear model which was trained to learn the ViCo embeddings. 
For this model, each entry in the co-occurrence matrix is the training data. In that case, to train it, how did you split the dataset i.e. co-occurrence matrix into train-test set? If you didn't split it and have used the entire dataset, how did you check if the model is learning and not memorizing? How did you decide to stop training after particular iterations?
Thanks,
Supriya

**Comment by BigRedT** [comment]
@supriya-gdptl these are great questions! 

Train/val/test splits are useful when there is a notion of generalization. However, while learning embeddings, there is no such notion of generalization (i.e. we are not using the learned parameters to compute embeddings for new words that were not seen during training). We simply care about getting embeddings that approximate the co-occurrence matrix as best as possible. So, we trained on the full co-occurrence matrix and optimized the objective until convergence. 

Having said that, it is possible that optimizing our chosen objective completely might not be ideal for the downstream task you are interested in. In that case, performance on the downstream task could be used for early stopping. In the paper, we wanted to learn embeddings in a task-agnostic manner and thus did not use such task-dependent early stopping.

**Comment by supriya-gdptl** [comment]
Thanks @BigRedT. This was helpful.
